src/libsyntax/parse/parser.rs

```diff
@@ -3668,6 +3665,62 @@ impl<'a> Parser<'a> {
         Ok(lhs)
     }
 
+    fn could_ascription_be_path(&self, node: &ast::ExprKind) -> bool {
+        self.token.is_ident() &&
+            if let ast::ExprKind::Path(..) = node { true } else { false } &&
+            !self.token.is_reserved_ident() &&           // v `foo:bar(baz)`
```

Comment:
Is this `v` intentional?

Reply:
It's meant to be an arrow pointing down 😬

Reply:
Oh, I see 😄I would be tempted just to put it on its own line, but this is fine too.